src/plugins/telemetryTable/collections/TableRowCollection.js

```diff
@@ -0,0 +1,482 @@
+                    } else {
+                        return true;
+                    }
+                });
+
+                this.emit('remove', removed);
+                this.removeTelemetryCollection(keyString);
+                delete this.telemetryObjects[keyString];
+            }
+
+            removeTelemetryCollection(keyString) {
+                if (this.telemetryCollections[keyString]) {
+                    this.telemetryCollections[keyString].destroy();
+                    this.telemetryCollections[keyString] = undefined;
+                    delete this.telemetryCollections[keyString];
+                }
+            }
+
+            getTelemetryProcessor(keyString, columnMap, limitEvaluator) {
+                return (telemetry) => {
+                    //Check that telemetry object has not been removed since telemetry was requested.
+                    if (!this.telemetryObjects[keyString]) {
+                        return;
+                    }
+
+                    // only cache realtime
+                    if (this.paused) {
+                        this.processFunctions.push(this.processTelemetryData.bind(this, telemetry, columnMap, keyString, limitEvaluator));
+                    } else {
+                        this.processTelemetryData(telemetry, columnMap, keyString, limitEvaluator);
+                    }
+                };
+            }
+
+            getTelemetryRemover() {
+                return (telemetry) => {
+                    // only cache realtime
+                    if (this.paused) {
+                        this.processFunctions.push(this.removeRowsByRowIds.bind(this, telemetry.map(JSON.stringify)));
+                    } else {
+                        this.removeRowsByRowIds(telemetry.map(JSON.stringify));
+                    }
+                };
+            }
+
+            processTelemetryData(telemetryData, columnMap, keyString, limitEvaluator) {
+                if (this.sortOptions === undefined) {
+                    throw 'Please specify sort options';
+                }
+
+                // anything coming from telemetry collection is in bounds and will be added (unless filtered)
+                let telemetryRows = telemetryData.map(datum => new TelemetryTableRow(datum, columnMap, keyString, limitEvaluator));
+
+                this.addRows(telemetryRows);
+            }
+
+            addRows(rows, skipEmit = false) {
+                let isFiltered = Object.keys(this.columnFilters).length > 0;
+                let rowsToAdd = !isFiltered ? rows : rows.filter(this.matchesFilters, this);
+
+                for (let row of rowsToAdd) {
+                    let index = this.sortedIndex(this.rows, row, 'sorted');
+                    this.rows.splice(index, 0, row);
+                }
+
+                if (!skipEmit && rowsToAdd.length > 0) {
+                    this.emit('add', rowsToAdd);
+                }
+            }
+
+            processCachedFunctions() {
```

Comment:
Let's maybe name this something like `processedPausedFunctions` or something.

Reply:
changed this to runDelayedActions